File: app/src/main/java/com/botpa/turbophotos/sync/SyncEventBus.java
package com.botpa.turbophotos.sync;

import androidx.lifecycle.MutableLiveData;

import java.util.concurrent.ConcurrentLinkedQueue;

public class SyncEventBus {

    private static SyncEventBus instance;

    private final ConcurrentLinkedQueue<SyncEvent> eventQueue = new ConcurrentLinkedQueue<>();
    private final MutableLiveData<Boolean> trigger = new MutableLiveData<>();

    public static synchronized SyncEventBus getInstance() {
        if (instance == null) instance = new SyncEventBus();
        return instance;
    }

    public ConcurrentLinkedQueue<SyncEvent> getEventQueue() {
        return eventQueue;
    }

    public MutableLiveData<Boolean> getTrigger() {
        return trigger;
    }

    public void postEvent(String command, String value) {
        eventQueue.add(new SyncEvent(command, value));
        trigger.postValue(true);
    }

    public void postEvent(String command, int value) {
        eventQueue.add(new SyncEvent(command, value));
        trigger.postValue(true);
    }

}

File: app/src/main/java/com/botpa/turbophotos/screens/sync/SyncEventBus.kt
package com.botpa.turbophotos.screens.sync

import androidx.lifecycle.MutableLiveData
import java.util.concurrent.ConcurrentLinkedQueue

class SyncEventBus {

    @JvmField val eventQueue: ConcurrentLinkedQueue<SyncEvent> = ConcurrentLinkedQueue<SyncEvent>()
    @JvmField val trigger: MutableLiveData<Boolean> = MutableLiveData<Boolean>()

    fun postEvent(command: String) {
        eventQueue.add(SyncEvent(command, 0)) //Use int as dummy value
        trigger.postValue(true)
    }

    fun postEvent(command: String, value: String) {
        eventQueue.add(SyncEvent(command, value))
        trigger.postValue(true)
    }

    fun postEvent(command: String, value: Int) {
        eventQueue.add(SyncEvent(command, value))
        trigger.postValue(true)
    }

    companion object {

        @JvmStatic
        @get:Synchronized
        var instance: SyncEventBus = SyncEventBus()

    }
}